nomad/structs/structs.go

```diff
@@ -850,6 +862,17 @@ type NetworkResource struct {
 	DynamicPorts  []Port // Dynamically assigned ports
 }
 
+func (n *NetworkResource) InitFields() {
+	// Ensure that an empty and nil slices are treated the same to avoid scheduling
+	// problems since we use reflect DeepEquals.
+	if len(n.ReservedPorts) == 0 && n.ReservedPorts != nil {
```

Comment:
`len(n.ReserverPorts) == 0` should suffice since len of empty and nil slice is 0